src/main/java/org/springframework/data/r2dbc/dialect/PostgresDialect.java

```diff
@@ -1,12 +1,25 @@
 package org.springframework.data.r2dbc.dialect;
 
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
 /**
  * An SQL dialect for Postgres.
  *
  * @author Mark Paluch
  */
 public class PostgresDialect implements Dialect {
 
+	private static final Set<Class<?>> SIMPLE_TYPES = new HashSet<>(
+			Arrays.asList(List.class, Collection.class, String[].class, UUID.class, URL.class, URI.class, InetAddress.class));
```

Comment:
Other primitive array types are supported via [`SimpleTypeHolder`](https://github.com/spring-projects/spring-data-commons/blob/master/src/main/java/org/springframework/data/mapping/model/SimpleTypeHolder.java#L45).

`supportsArrayColumns` is here to decide what to do with `Collection`-typed properties. We fail fast for dialects that do not have support for array types. What would you suggest?

Reply:
After looking at the rest of the code. Shouldn't everything just work without the `String[].class` here? If we detect a collection/ we check `supportsArrayColumns` and essentially check that it contains a simple type, which `String` should be already.

At least that is the behavior that I'd expect: if a driver supports arrays we assume it supports it for all simple types.

While writing this another corner case becomes obvious: since some arrays are considered simple types we would try to work with two-dimensional arrays for those types, but not for higher dimensions. And if that works is open to debate^H^H^H^H^H^H testing. 

After all that rambling my suggestion/expectation:
1. remove `String[].class` here
2. remove the various array types from the simple types.
3. tweak the logic that it works based on `supportsArrayColumns` for collections and arrays containing simple types.

Reply:
> After looking at the rest of the code. Shouldn't everything just work without the String[].class here?

Unfortunately, that's not the case. `MappingContext` creates an entity for `String[]` typed properties. Not sure whether you're already heading the direction to inspect the component type of properties. This could work to unwrap the component type and assume that arrays are just a flavor of simple types.

> At least that is the behavior that I'd expect: if a driver supports arrays we assume it supports it for all simple types.

This is true for Postgres but different for other databases such as Informix. We will deal with this assumption later on.

> … would try to work with two-dimensional arrays for those types, but not for higher dimensions. 

We face two issues here:
1. Conversion of `Collection<T>` to `T[]`
2. Support of `T[][]`

Our mapping layer should be able to convert collections to arrays for single-dimensional arrays in the first place. Using multi-dimensional arrays requires native type usage (e.g. `String[][]` instead of `List<List<String>>`).

I will apply the single-dimension conversion change.